### ISSUE TITLE
Add thread name prefix option for gthread worker

### ISSUE THREAD

**Issue by awiddersheim** [benoitc/gunicorn#1991]
When using the gthread worker, it would be nice to be able to pass `thread_name_prefix` to the underlying `ThreadPoolExecutor`. This would help in logging situations. For example my logs look like this:

```
2019-02-20 13:35:17.762 - [INFO] [18] (Foo_0): In custom threadpool
2019-02-20 13:35:18.705 - [INFO] [18] (ThreadPoolExecutor-0_0): Back in gunicorn
```

Something more like this would be a bit nicer:

```
2019-02-20 13:35:17.762 - [INFO] [18] (Foo_0): In custom threadpool
2019-02-20 13:35:18.705 - [INFO] [18] (Gunicorn_0): Back in gunicorn
```

Some drawbacks to this idea are that `thread_name_prefix` was only made an option in Python >=3.6 (https://github.com/python/cpython/commit/50abe877ee6f50ebd9cfe228d314220e071fa3c6) so this would be a version specific option for now. Also, if the underlying gthread implementation moves away from `ThreadPoolExecutor`s  in the future for any reason to something where this prefix can no longer bet set, that would be annoying.

Curious to know thoughts on adding this option and any suggestions on implementing. I would fine with doing the legwork.

**Comment by tilgovi** [comment]
An alternative might be to refactor the threaded worker very lightly so that it's easy to override the pool creation. Then you could subclass the worker for your own needs.

**Comment by tilgovi** [comment]
Maybe a good question we can ask is whether there are other customizations of the pool that users might want.

**Comment by awiddersheim** [comment]
@tilgovi I like the idea of refactoring the worker class to take a `pool_kwargs` argument that one can supply in the config or something as a first step.

**Comment by awiddersheim** [comment]
Are you able to pass arbitrary settings? Adding something like `pool_kwargs` might be tightly coupling things a bit. If it is not possible to pass arbitrary things using `gunicorn.py` than the next best looking thing would be something like similar to `secure_scheme_headers` (https://github.com/benoitc/gunicorn/blob/master/gunicorn/config.py#L1159).

**Comment by awiddersheim** [comment]
Hrm I suppose you could tie into `post_worker_init
` as well (http://docs.gunicorn.org/en/stable/settings.html#post-worker-init).

**Comment by tilgovi** [comment]
I was even thinking you could just use `--worker-class mypackage.MyWorker` and subclass the threaded worker, overriding `init_process` or some other method into which we factor out the pool creation. https://github.com/benoitc/gunicorn/blob/master/gunicorn/workers/gthread.py#L88

**Comment by awiddersheim** [comment]
@tilgovi something like this?

https://github.com/benoitc/gunicorn/compare/master...awiddersheim:gthread_pool_subclass?expand=1

**Comment by tilgovi** [comment]
Yup! That's what I was thinking.

**Comment by awiddersheim** [comment]
Fixed in #1992. Closing.
